lib/SILOptimizer/SILCombiner/SILCombinerCastVisitors.cpp

```diff
@@ -452,6 +452,10 @@ SILCombiner::visitThickToObjCMetatypeInst(ThickToObjCMetatypeInst *TTOCMI) {
   if (CastOpt.optimizeMetatypeConversion(TTOCMI, MetatypeRepresentation::Thick))
     MadeChange = true;
 
+  if (auto *OCTTMI = dyn_cast<ObjCToThickMetatypeInst>(TTOCMI->getOperand())) {
+    TTOCMI->replaceAllUsesWith(OCTTMI->getOperand());
```

Comment:
I think you need to use the SILCombiner::replaceAllUsesWith method, to make sure the wordlist is updated. Anytime one mutates anything in SILCombiner, one needs to do it via those function to ensure that the worklist is updated appropriately.

Reply:
I think you mean `SILCombiner::replaceInstUsesWith` ? I didn't add it because most of the SILCombiner seems to be using the `replaceAllUsesWith` + `eraseInstFromFunction` pattern. And using `SILCombiner::replaceInstUsesWith` and `eraseInstFromFunction` seems to add the operands of the instruction to be deleted twice into the worklist

Reply:
Ok. Two thoughts:

1. The good thing is we only add things to the worklist once IIRC since it is a set vector... so even if one did it by mistake, we wouldn't get bad behavior = p. So at least if this was done in other areas, we are cool.
2. SGTM. I think it is counter-intuitive to me due to the way we have started using InstModCallbacks where one specifies all such operations and you have to use the callbacks for everything!